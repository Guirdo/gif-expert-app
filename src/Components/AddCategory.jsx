import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bootstrap';

const AddCategory = ({setCategories}) => {

    const [inputValue,setInputValue] = useState('');
    
    const handleInputChange = (e) =>{
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(c => [inputValue,...c]);
            setInputValue('');
        }

    }

    return (
        <Form onSubmit={handleSubmit}>
            <Form.Control 
                type="text"
                value={inputValue}
                placeholder="Search anything you want"
                onChange={ handleInputChange }
            />
        </Form>
    )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory;
