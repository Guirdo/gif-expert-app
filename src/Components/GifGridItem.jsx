import { Col,Image } from 'react-bootstrap';

export const GifGridItem = ( { title,url } ) => {
    return (
        <Col sm={10} md={3} lg={3} className="mx-1 mb-4 p-2 bg-white animate__animated animate__fadeIn border border-dark rounded">
            <Image 
                alt={title}
                src={url}
                fluid
            />
            <p className="mt-3 text-center">{title}</p>
        </Col>
    )
}
