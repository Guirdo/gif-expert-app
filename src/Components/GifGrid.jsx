import { Col,Row } from 'react-bootstrap';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: images,loading } = useFetchGifs(category);

    return (
        <Col>
            <h3 className="mt-2 animate__animated animate__flipInX">{category}</h3>

            {loading && <p className="animate__animated animate__bounce">LOADING...</p>}

            <Row className="justify-content-around mt-3">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </Row>
        </Col>
    )
}
