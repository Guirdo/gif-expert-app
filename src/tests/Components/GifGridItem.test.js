import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../Components/GifGridItem";

describe('Testing GifGridItem',()=>{

    const title = 'A sample title';
    const url = 'https://localhost/something.jpg';

    const wrapper = shallow(
        <GifGridItem 
            title={title}
            url={url}
        />
    );

    test('should render GifGridItem correctly',()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a paragraph with the title',()=>{
        const p = wrapper.find('p');

        expect(p.text()).toBe(title);
    });

    test('should have the url and alt like props',()=>{
        const img = wrapper.find('Image');
        
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('should have animate__fadeIn',()=>{
        const div = wrapper.find('Col');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
    });

});