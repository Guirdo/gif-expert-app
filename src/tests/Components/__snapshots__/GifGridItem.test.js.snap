// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should render GifGridItem correctly 1`] = `
<Col
  className="mx-1 mb-4 p-2 bg-white animate__animated animate__fadeIn border border-dark rounded"
  lg={3}
  md={3}
  sm={10}
>
  <Image
    alt="A sample title"
    fluid={true}
    rounded={false}
    roundedCircle={false}
    src="https://localhost/something.jpg"
    thumbnail={false}
  />
  <p
    className="mt-3 text-center"
  >
    A sample title
  </p>
</Col>
`;
